editor/components/block-list/insertion-point.js

```diff
@@ -11,28 +17,51 @@ import {
 	getBlockInsertionPoint,
 	isBlockInsertionPointVisible,
 	getBlockCount,
+	getBlock,
+	isTyping,
 } from '../../store/selectors';
+import {
+	insertDefaultBlock,
+	startTyping,
+} from '../../store/actions';
 
-function BlockInsertionPoint( { showInsertionPoint } ) {
-	if ( ! showInsertionPoint ) {
-		return null;
-	}
+function BlockInsertionPoint( { showInsertionPoint, showInserter, index, layout, rootUID, ...props } ) {
+	const onClick = () => {
+		props.insertDefaultBlock( { layout }, rootUID, index );
+		props.startTyping();
```

Comment:
Should `startTyping` be an effect of `insertDefaultBlock`?

Reply:
While we use it this way in general, I wonder if it makes sense conceptually. A third-party plugin would want to trigger `insertDefaultBlock` programmatically without impact on the writing flow.